modules/core/src/smithy4s/Document.scala

```diff
@@ -63,6 +63,8 @@ sealed trait Document extends Product with Serializable {
       value.map { case (k, v) => k + "=" + v.show }.mkString("{", ", ", "}")
   }
 
+  // Nests this document under the given key.
```

Comment:
Let's turn that into a proper bit of scaladoc

Reply:
c243c7a1f748ae493bbec70d73f66751205d000c